src/Objs/Redirect/RedirectComponent.tsx

```diff
@@ -11,22 +12,23 @@ import { Redirect } from './RedirectObjClass'
 provideComponent(Redirect, ({ page }) => {
   const requireLogin = page.get('requireLogin')
   const link = page.get('link')
-  const url = link && urlFor(link)
 
   React.useEffect(() => {
     if (requireLogin && !isUserLoggedIn()) {
       ensureUserIsLoggedIn()
       return
     }
 
-    if (!link || !url) return
+    load(() => link && urlFor(link)).then((url) => {
```

Comment:
* [ ] FYI For simpler code  the `!link` condition could be a guard at the beginning of the effect callback